apps/passport-client/src/password.ts

```diff
@@ -9,3 +16,28 @@ export const checkPasswordStrength = (password: string): boolean => {
 };
 
 export const PASSWORD_MINIMUM_LENGTH = 8;
+
+// For when the user has not set a password yet, and wants to add one
+export const setPassword = async (password: string, dispatch: Dispatcher) => {
+  const crypto = await PCDCrypto.newInstance();
+  const currentEncryptionKey = loadEncryptionKey();
+  const { salt: newSalt, key: newEncryptionKey } =
+    await crypto.generateSaltAndEncryptionKey(password);
+  const res = await updateBlobKeyForEncryptedStorage(
+    currentEncryptionKey,
+    newEncryptionKey,
+    newSalt
+  );
+
+  if (!res.success) {
+    throw new Error(`Request failed with message ${res.error}`);
+  }
+
+  dispatch({
+    type: "change-password",
+    newEncryptionKey,
+    newSalt
+  });
+  // to make sure the salt is uploaded properly
```

Comment:
What does this comment mean?  The salt is uploaded inside of updateBlobKeyForEncryptedStorage.

Reply:
this was copied from `UpgradePaswordModal.tsx`, but you're right that it's unnecessary. Removing.